sdk/keyvault/Azure.Security.KeyVault.Keys/src/JsonWebKey.cs

```diff
@@ -138,10 +183,96 @@ public JsonWebKey(Aes aesProvider)
         #endregion
 
         /// <summary>
-        /// HSM Token, used with "Bring Your Own Key"
+        /// HSM Token, used with "Bring Your Own Key".
         /// </summary>
         public byte[] T { get; set; }
 
+        /// <summary>
+        /// Converts this <see cref="JsonWebKey"/> of type <see cref="KeyType.Octet"/> to an <see cref="Aes"/> object.
+        /// </summary>
+        /// <returns>An <see cref="Aes"/> object.</returns>
+        /// <exception cref="InvalidOperationException">This key is not of type <see cref="KeyType.Octet"/> or <see cref="K"/> is null.</exception>
+        public Aes ToAes()
+        {
+            if (KeyType != KeyType.Octet)
+            {
+                throw new InvalidOperationException("key is not an octet key");
+            }
+
+            if (K is null)
+            {
+                throw new InvalidOperationException("key does not contain a value");
+            }
+
+            Aes key = Aes.Create();
+            if (key != null)
+            {
+                key.Key = K;
+            }
+
+            return key;
+        }
+
+        /// <summary>
+        /// Converts this <see cref="JsonWebKey"/> of type <see cref="KeyType.EllipticCurve"/> or <see cref="KeyType.EllipticCurveHsm"/> to an <see cref="ECDsa"/> object.
+        /// </summary>
+        /// <param name="includePrivateParameters">Whether to include private parameters.</param>
+        /// <returns>An <see cref="ECDsa"/> object.</returns>
+        /// <exception cref="InvalidOperationException">This key is not of type <see cref="KeyType.EllipticCurve"/> or <see cref="KeyType.EllipticCurveHsm"/>, or one or more key parameters are invalid.</exception>
+        public ECDsa ToECDsa(bool includePrivateParameters = false)
+        {
+            if (KeyType != KeyType.EllipticCurve && KeyType != KeyType.EllipticCurveHsm)
+            {
+                throw new InvalidOperationException("key is not an EC or EC-HSM type");
```

Comment:
>"key is not an EC or EC-HSM type" [](start = 52, length = 33)

We should probably change the message to reflect the Enum Values rather than the string Values (although personally I think we should change the Enum values to be closer to the string values)